x/oracle/types/msgs.go

```diff
@@ -168,3 +172,115 @@ func (msg MsgDelegateFeedConsent) ValidateBasic() error {
 
 	return nil
 }
+
+// NewMsgUpdateParams will creates a new MsgUpdateParams instance
+func NewMsgUpdateParams(authority, title, description string, keys []string, changes Params) *MsgGovUpdateParams {
+	return &MsgGovUpdateParams{
+		Title:       title,
+		Description: description,
+		Authority:   authority,
+		Keys:        keys,
+		Changes:     changes,
+	}
+}
+
+// Type implements Msg interface
+func (msg MsgGovUpdateParams) Type() string { return sdk.MsgTypeURL(&msg) }
+
+// String implements the Stringer interface.
+func (msg MsgGovUpdateParams) String() string {
+	out, _ := yaml.Marshal(msg)
+	return string(out)
+}
+
+// GetSignBytes implements Msg
+func (msg MsgGovUpdateParams) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(&msg)
+	return sdk.MustSortJSON(bz)
+}
+
+// GetSigners implements Msg
+func (msg MsgGovUpdateParams) GetSigners() []sdk.AccAddress {
+	return checkers.Signers(msg.Authority)
+}
+
+// ValidateBasic implements Msg
+func (msg MsgGovUpdateParams) ValidateBasic() error {
+	if err := checkers.ValidateProposal(msg.Title, msg.Description, msg.Authority); err != nil {
+		return err
+	}
+
+	for _, key := range msg.Keys {
```

Comment:
So if one part of the message doesn't pass validation, none of it goes through right?

Reply:
No that's good, don't want partials going through